Changed traversion of the messages...

sauin opened the pull request:
… during blob restoration if snapshot was deleted

Commit:
Added test

Commit:
Added fixes

Now branch is deployed but upgraded commit is not correct

Commit:
changed traversion of the messages during blob restoration if snapshot was deleted

Commit:
added tests

SilkovAlexander commented on v1_x/git-remote-gosh/src/git_helper/fetch/mod.rs:
```diff
@@ -204,16 +213,22 @@ where
+                        branches.insert(branch.name);
+                    }
+                } else {
+                    branches.insert(branch.to_owned());
```
Isn't it the same as ^

SilkovAlexander commented on v1_x/git-remote-gosh/src/git_helper/fetch/mod.rs:
```diff
@@ -204,16 +213,22 @@ where
-                        None
-                    ).await?;
-                    branches.insert(branch.name);
+                branches.insert(onchain_commit.branch.clone());
```
->

SilkovAlexander commented on v3_x/git-remote-gosh/src/blockchain/snapshot/load/diffs/iterator.rs:
```diff
@@ -133,6 +133,10 @@ impl DiffMessagesIterator {
+                let snapshot_contract =
+                    GoshContract::new(original_snapshot.clone(), crate::abi::SNAPSHOT);
+                let snapshot_is_active = snapshot_contract.is_active(client).await?;
+                tracing::trace!("snap={original_snapshot} is {}", if snapshot_is_active { "ACTIVE" } else { "NON_ACTIVE" });
```
v3 version of this file has more logs than others, I suggest add this extra tracing strings to at least the last version

andrew-kurochkin reviewed (changes requested):
Please-please add comments to tests

Commit:
Merge branch 'fixes/bug_with_branch' into bugfix/restore-blob-through-deleted-branch

Commit:
Merge branch 'fixes/bug_with_branch' into bugfix/restore-blob-through-deleted-branch

Commit:
Updated and moved test

Commit:
described test 23

Commit:
Fixed rights and debug ci

Commit:
rename test 23 to 23_1

Commit:
Merge branch 'bugfix/restore-blob-through-deleted-branch' of github.com:gosh-sh/gosh into bugfix/restore-blob-through-deleted-branch

Commit:
Returned build to ci

Commit:
Merge branch 'dev' into bugfix/restore-blob-through-deleted-branch

Commit:
Merge branch 'bugfix/restore-blob-through-deleted-branch' of github.com:gosh-sh/gosh into bugfix/restore-blob-through-deleted-branch

Commit:
Updated rust version

Commit:
fix naming for bins

Commit:
fix naming for bins

Commit:
fix naming for bins

Commit:
fix naming for bins

Commit:
fix naming for bins

Commit:
fix naming for bins

Commit:
fix naming for bins

Commit:
fix naming for bins

Commit:
fix naming for bins

Commit:
fix naming for bins

Commit:
fix naming for bins

Commit:
fix naming for bins

Commit:
fix naming for bins

Commit:
fix naming for bins

Commit:
fix naming for bins

Commit:
fix naming for bins

Commit:
fix naming for bins

Commit:
fix naming for bins

Commit:
fix naming for bins

Commit:
ci

Commit:
Fixed bug with fetch after upgrade

Commit:
ci

Commit:
ci

Commit:
ci

Commit:
ci

Commit:
ci

Commit:
ci

Commit:
ci

Commit:
ci

Commit:
ci

Commit:
ci

Commit:
ci

Commit:
ci

Commit:
ci

Commit:
ci

Commit:
ci

Commit:
Only upgrade tests

Commit:
ci

Commit:
fix fetch

fetch command can duplicate thats why it's important search for another version not only in ancestors

Commit:
ci

Commit:
ci

Commit:
changelog and versions

Commit:
Merge remote-tracking branch 'origin/dev' into bugfix/restore-blob-through-deleted-branch

Commit:
Fixed default config in install.sh

Commit:
Prepare CI for fast release

Commit:
fix install sh for old releases

Commit:
Added condition to run web tests
